pkgs/build-support/bintools-wrapper/add-hardening.sh

```diff
@@ -42,8 +42,13 @@ for flag in "${!hardeningEnableMap[@]}"; do
             || " ${params[*]} " =~ " -r " \
             || " ${params[*]} " =~ " -Ur " \
             || " ${params[*]} " =~ " -i ") ]]; then
-        if (( "${NIX_DEBUG:-0}" >= 1 )); then echo HARDENING: enabling LDFlags -pie >&2; fi
-        hardeningLDFlags+=('-pie')
+        if (( @isStatic@ )); then
+          if (( "${NIX_DEBUG:-0}" >= 1 )); then echo HARDENING: enabling LDFlags -pie --no-dynamic-linker >&2; fi
```

Comment:
Maybe

```bash
hardeningLDFlags+=('-pie')
if (( @isStatic@ )); then
    hardeningLDFlags+=('--no-dynamic-linker')
fi
```

Reply:
Then there's no easy way to print the right message.  If I introduce a variable, it's global, so I don't want to do that.

Reply:
Can we not just print `hardeningLDFlags`?

Reply:
Like remove all the conditional prints and just print the final value at the end